### PR TITLE
[SASS-82] GTM tag added

### PR DESCRIPTION
**Pull request by dnyuar**

### CODE REVIEW

File: app/views/headerFooterTemplate/Head.scala.html [hunk 1]
```diff
@@ -14,10 +14,21 @@
...
 <!--[if lte IE 8]><link href='@controllers.routes.Assets.versioned("stylesheets/application-ie-8.css")' rel="stylesheet" type="text/css" /><![endif]-->
 <!--[if gt IE 8]><!--><link href='@controllers.routes.Assets.versioned("stylesheets/application.css")' media="screen" rel="stylesheet" type="text/css" /><!--<![endif]-->
+
+@if(appConfig.googleTagManagerId != "N/A") {
+  <script type="text/javascript" src='@controllers.routes.Assets.versioned("javascripts/google-tag-manager.js")'></script>
+
+  <noscript>
+     <iframe src="https://www.googletagmanager.com/ns.html?id=@appConfig.googleTagManagerId" height="0" width="0" style="display: none; visibility: hidden"></iframe>
+  </noscript>
+  }
```
Comment (juamalBlackman) [inline app/views/headerFooterTemplate/Head.scala.html:34]: new line

File: app/views/headerFooterTemplate/Head.scala.html [hunk 2]
```diff
@@ -14,10 +14,21 @@
...
-@(headBlock: Option[Html] = None)
+@(headBlock: Option[Html] = None)(implicit appConfig: FrontendAppConfig)
+
 @headBlock
 <!--[if lte IE 8]><script src='@controllers.routes.Assets.versioned("javascripts/html5shiv.min.js")'></script><![endif]-->
 <!--[if lte IE 8]><link href='@controllers.routes.Assets.versioned("stylesheets/application-ie-8.css")' rel="stylesheet" type="text/css" /><![endif]-->
 <!--[if gt IE 8]><!--><link href='@controllers.routes.Assets.versioned("stylesheets/application.css")' media="screen" rel="stylesheet" type="text/css" /><!--<![endif]-->
+
+@if(appConfig.googleTagManagerId != "N/A") {
+  <script type="text/javascript" src='@controllers.routes.Assets.versioned("javascripts/google-tag-manager.js")'></script>
```
Comment (juamalBlackman) [inline app/views/headerFooterTemplate/Head.scala.html:29]: this is not picking up the file 

https://github.com/hmrc/charities-frontend/commit/8364074094604fbf81109933066c4b367ff8f19d
the javascript file is in public in this example

Comment (TheDeadlyPianist) [reply app/views/headerFooterTemplate/Head.scala.html:29]: It's worth noting that these services are set up slightly different, so the way it works will be a little different.

Comment (robert-webb-hmrc) [reply app/views/headerFooterTemplate/Head.scala.html:29]: If there are still issues afterwards, here's DDCNLS's guide which might help. The implementations are slightly different but all the code should be the same. https://confluence.tools.tax.service.gov.uk/display/DDCNLS/Adding+GTM+to+services

File: app/views/headerFooterTemplate/Head.scala.html [hunk 3]
```diff
@@ -14,10 +14,24 @@
...
+@if(appConfig.googleTagManagerId != "N/A") {
+<script type="text/javascript" src='@controllers.routes.Assets.versioned("javascript/google-tag-manager.js")'></script>
+
+<noscript>
+   <iframe src="https://www.googletagmanager.com/ns.html?id=@appConfig.googleTagManagerId"
+         height="0" width="0" style="display: none; visibility: hidden">
+   </iframe>
+</noscript>
+
+}
```
Comment (juamalBlackman) [inline app/views/headerFooterTemplate/Head.scala.html:37]: newLine

File: public/javascript/google-tag-manager.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+
+// Google Tag Manager
+(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start': new Date().getTime(),event:'gtm.js'});
+    var f=d.getElementsByTagName(s)[0], j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';
+    j.async=true;j.src= 'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
+    })(window,document,'script','dataLayer','GTM-TSFTCWZ');
+// End Google Tag Manager
```
Comment (juamalBlackman) [inline public/javascript/google-tag-manager.js:7]: newLine

**Pull request merged**